db/user.go

```diff
@@ -221,3 +222,45 @@ func SyncUsers(course *model.Course, newUsers map[string]*model.User,
 
     return syncResult, nil;
 }
+
+// Insert method that takes a course, email [] string and resolves based on *, roles, emails to all emails
+func ResolveUsers(course *model.Course, emails []string) ([]string, error) {
+    if (backend == nil) {
+        return nil, fmt.Errorf("Database has not been opened.");
+    }
+    emailSet := map[string]any{};
+    roleSet := map[model.UserRole]any{};
+    // check for emails, roles, and * (all)
+    for _, email := range emails {
+        if (strings.Contains(email, "@") || (email == "")) {
```

Comment:
It seems to me like empty strings should be thrown away.
It looks like these make it all the way through.